solver/exporter.go

```diff
@@ -117,6 +117,13 @@ func (e *exporter) ExportTo(ctx context.Context, t CacheExporterTarget, opt Cach
 			return nil, err
 		}
 
+		if e.edge != nil {
+			op, ok := e.edge.op.(*sharedOp)
+			if ok && op != nil && op.st != nil {
+				ctx = withAncestorCacheOpts(ctx, op.st)
```

Comment:
Would it make sense to cache this per `st`?

Reply:
Maybe, you'd be using worst case O(n^2) memory rather than O(n^2) time (where n is the number of ancestor vertices in the DAG of the cache ref being exported) though, so more of a tradeoff than strictly better.

I'm hesitant to add more complexity to all of this until proven a bottleneck though (pretty sure you'd need a truly enormous DAG for this to matter). Let me know what you think.